### PR TITLE
Fix duplicated benchmarks

### PR DESCRIPTION
**Pull request by terceiro**

### CODE REVIEW

**Comment by terceiro** [comment]
reviewing commit-by-commit will make a lot more sense

File: benchmarks/tasks.py [hunk 1]
```diff
@@ -39,32 +78,33 @@ def set_testjob_results(self, testjob):
     if not testjob.initialized:
         testjob.testrunnerclass = tester.get_result_class_name(testjob.id)
         testjob.initialized = True
-        testjob.save()
```
Comment (mwasilew) [inline benchmarks/tasks.py]: I don't understand this bit. If the testjob.initialized isn't saved to db it's lost and the condition 'if not testjob.initialized' will be met as the default is False. Am I missing something here?

File: benchmarks/tasks.py [hunk 2]
```diff
@@ -39,32 +78,33 @@ def set_testjob_results(self, testjob):
...
         tester = getattr(testminer, testjob.testrunnerclass)(
             testjob.testrunnerurl, username, password
         )
 
     if testjob.status not in ["Complete", "Incomplete", "Canceled"]:
-        logger.debug("Saving job({0}) status: {1}".format(testjob.id, testjob.status))
-        testjob.save()
+        logger.debug("Job({0}) status: {1}".format(testjob.id, testjob.status))
         return
```
Comment (mwasilew) [inline benchmarks/tasks.py]: Will it return 'None'/'False'? There is a condition checking what is returned from this function

File: benchmarks/tasks.py [hunk 3]
```diff
@@ -39,32 +78,33 @@ def set_testjob_results(self, testjob):
...
 
     logger.debug("Calling testminer")
     logger.debug("Tester class:{0}".format(tester.__class__.__name__))
     logger.debug("Testjob:{0}".format(testjob.id))
+
+    # already got test results; bail out
+    if testjob.result.data.count() > 0:
```
Comment (mwasilew) [inline benchmarks/tasks.py]: I'm not sure this is a correct assumption. The Result object might have data from other TestJobs. So we would only get results from the first completed TestJob that produces useful results.

Comment (mwasilew) [inline benchmarks/tasks.py]: Looks almost OK. I'm not really sure what happens if there are 2 testjobs contributing to the same Result object. Will try it locally.

**Comment by mwasilew** [comment]
Looks almost OK. I'm not really sure what happens if there are 2 testjobs contributing to the same Result object. Will try it locally.

**Pull request closed without merging**